src/jit/IntMath32Inl.h

```diff
@@ -1122,6 +1125,12 @@ static bool emitCompare(sljit_compiler* compiler, Instruction* instr)
             jump = sljit_emit_cmp(compiler, type, params[0].arg, params[0].argw, params[1].arg, params[1].argw);
```

Comment:
I think `*operand != VARIABLE_SET_PTR(nullptr)` is possible here as well.  Maybe we should add it to the condition (and make a test for it)